rustls/src/polyfill.rs

```diff
@@ -1,5 +1,12 @@
 /// Non-panicking `let (nonce, ciphertext) = ciphertext.split_at(...)`.
 // TODO(XXX): remove once stabilized in https://github.com/rust-lang/rust/issues/119128
+#[cfg_attr(
+    any(
+        not(any(feature = "aws-lc-rs", feature = "ring")),
+        not(feature = "std")
+    ),
+    allow(dead_code)
+)]
```

Comment:
This boolean logic is a bit gnarly so I'd also be open to just slapping a `#[allow(dead_code)]` on this unconditionally. I thought I'd try to do it the right way first.

Reply:
Something like `#[allow(dead_code)] // Complicated conditional compilation guards elided` sounds good to me.

Reply:
WFM. Applied verbatim.